### DISCUSSION TITLE
Help with automatic struct change in neovim

### DISCUSSION THREAD

**Discussion by h0pes** [mrcjkb/rustaceanvim#120] [Q&A]
Hi,
I'm posting it here too, to seek some possible good advice from valuable members. Not saying it is related to this plugin at all!
I'm getting weird automatic struct changes upon `Insert Line` 
![2024-01-05_08-57_rust_struct_definition](https://github.com/mrcjkb/rustaceanvim/assets/25418553/24790b30-e635-4188-88f1-74b190097140)
![2024-01-05_08-56_struct_field_deleted](https://github.com/mrcjkb/rustaceanvim/assets/25418553/f6eaff5b-0466-4f2f-a9ac-29bdf37b5693)

Thread with detailed explanation:

[stackexchange thread](https://vi.stackexchange.com/questions/43787/debug-help-auto-deletion-first-struct-field-upon-inserting-new-line-in-buffer-w)

**Comment by mrcjkb** [comment]
Hey 👋 

This plugin doesn't do any auto-formatting.
But some Neovim distros set up autocommands to do that.

Off the top of my head, I would guess that your distro or a plugin has set up multiple autocommands for formatting, and they could be conflicting with each other.

I would suggest you do the following to find the cause:

- Follow this plugin's [troubleshooting guide](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#minimal-config), starting from a minimal config. Add your settings and plugins you use until you can reproduce the issue.
- Use [`:h autocmd`](https://neovim.io/doc/user/autocmd.html) to find out which autocommands could be causing the behaviour.

**Comment by h0pes** [comment]
Thanks God, I found the culprit. And it seems someone already opened an issue some time ago. [tree-setter](https://github.com/filNaj/tree-setter/issues/17)
